packages/postcss-normalize-whitespace/src/__tests__/index.js

```diff
@@ -56,3 +56,11 @@ test(
     'h1{width:calc(10px - 100px/var(--test))}'
   )
 );
+
+test(
+  'should not trim spaces inside of var function',
+  withDefaultPreset(
+    'div{border-radius:10px var(--foobar, )}',
+    'div{border-radius:10px var(--foobar, )}'
+  )
+);
```

Comment:
Let's add tests:
```
div{var(--my-var, var(--my-background, pink, ))}
div{calc(var(--text-xxxl, ) * var(--text-scale-ratio-up, ))}
```
Also same tests and instead `var` use `env` (https://developer.mozilla.org/en-US/docs/Web/CSS/env)

Reply:
Please add tests as described here

Reply:
sure 👍

Reply:
@evilebottnawi
removing whitespace inside of `calc` is not valid I guess.
cause of this

`div{border-radius:calc(10px var(--foobar, ))}`
⬇️
`div{border-radius:calc(10pxvar(--foobar, ))}`

And this minification is not valid. So I guess removing whitespace from `env, calc, var` is not safe.
even removing `before` and `after` of these are not safe either.

Reply:
@evilebottnawi This would be the fix if allowing removing of spaces inside `calc` but not inside `env and var`

```js
function reduceCalcWhitespaces(node) {
  if (node.type === "space") {
    node.value = " ";
  } else if (node.type === "function") {
    if (!["var", "env"].includes(node.value.toLowerCase())) {
      node.before = node.after = "";
    }
  }
}

function reduceWhitespaces(node) {
  if (node.type === "space") {
    node.value = "  ";
  } else if (node.type === "div") {
    node.before = node.after = "";
  } else if (node.type === "function") {
    if (!["var", "env"].includes(node.value.toLowerCase())) {
      node.before = node.after = "";
    }
    if (node.value.toLowerCase() === "calc") {
      valueParser.walk(node.nodes, reduceCalcWhitespaces);
      return false;
    }
  }
}
```

Reply:
```
div{border-radius:calc(10px var(--foobar, ))}
```

It is invalid CSS, so as i written early:

invalid CSS === invalid minification

Reply:
Yeah I doubted that too
Ok i will do trimming of spaces for all expected inside the var and env